src/Squirrel/UpdateManager.ApplyReleases.cs

```diff
@@ -317,16 +318,72 @@ Task<string> installPackageToAppDir(UpdateInfo updateInfo, ReleaseEntry release,
                     target.Create();
 
                     this.Log().Info("Writing files to app directory: {0}", target.FullName);
-                    await ReleasePackage.ExtractZipForInstall(
-                        Path.Combine(updateInfo.PackageDirectory, release.Filename),
-                        target.FullName,
-                        rootAppDirectory,
-                        progressCallback);
+
+                    string fullPackageDir = Path.Combine(updateInfo.PackageDirectory, release.Version.ToString());
+                    string fullPackageNuget = Path.Combine(updateInfo.PackageDirectory, release.Filename);
+                    
+                    // Check if the release to apply is full nuget package.
+                    if (updateInfo.ReleasesToApply.Count == 1 && updateInfo.ReleasesToApply[0].IsDelta == false) {
+                        // This might happen if the previous release applying failed.
+                        if (Directory.Exists(fullPackageDir)) {
+                            await Utility.DeleteDirectory(fullPackageDir);
+                        }
+
+                        // Extract nuget in the packages directory to have it prepared for deltas applying in the future.
+                        await ReleasePackage.extractZipWithEscaping(fullPackageNuget, fullPackageDir, progressCallback);
+
+                        // Create nuget package which contains only nuget metadata without app.
+                        // Can be used later by functions which expects nupkg instead of unpacked dir.
+                        // It overwrites downloaded full package to save some space.
+                        using (ZipFile zip = new ZipFile()) {
```

Comment:
The same snippet as in DeltePackage.cs. Might be a good idea to reuse the same function. But not if this would require creating a new file. Just pointing out, if there's a common place that you could move it to...

Reply:
Good catch. Moved to the _ReleasePackage.cs_ file for better reusability and maintanance.